packages/esm-sms-app/src/providers/add-config-form/provider-config-form.component.tsx

```diff
@@ -0,0 +1,346 @@
+import { saveConfig } from '../../api/providers.resource';
+import { useProviderConfigurations } from '../../hooks/useProviderConfigurations';
+import { closeOverlay } from '../../hooks/useOverlay';
+import styles from './provider-config-form.scss';
+
+interface AddProviderConfigProps {
+  providerName?: string;
+  templateName?: string;
+  maxRetries?: number;
+  splitHeader?: string;
+  splitFooter?: string;
+  excludeLastFooter?: boolean;
+  autoScript?: string;
+  props?: Array<Prop>;
+}
+
+const addProviderConfigFormSchema = z.object({
+  providerName: z.string(),
+  templateName: z.string(),
+  maxRetries: z.number(),
+  splitHeader: z.string(),
+  splitFooter: z.string(),
+  excludeLastFooter: z.boolean(),
+  autoScript: z.string().optional(),
+  dynamicFields: z.record(z.string(), z.string()),
+});
+
+type AddProviderConfigFormData = z.infer<typeof addProviderConfigFormSchema>;
+
+const AddProviderConfigForm: React.FC<AddProviderConfigProps> = ({
+  providerName,
+  templateName,
+  excludeLastFooter,
+  splitFooter,
+  splitHeader,
+  maxRetries,
+  autoScript,
+  props,
+}) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const { mutateConfigs, providerConfigurations } = useProviderConfigurations();
+  const { templates, isLoadingTemplates, error } = useProviderConfigTemplates();
+
+  const {
+    handleSubmit,
+    control,
+    watch,
+    setValue,
+    formState: { isSubmitting, isDirty },
+  } = useForm<AddProviderConfigFormData>({
+    mode: 'all',
+    resolver: zodResolver(addProviderConfigFormSchema),
+    defaultValues: {
+      providerName: providerName ? providerName : undefined,
+      templateName: templateName ? templateName : undefined,
+      maxRetries: maxRetries ? maxRetries : 0,
+      splitHeader: splitHeader ? splitHeader : undefined,
+      splitFooter: splitFooter ? splitFooter : undefined,
+      excludeLastFooter: excludeLastFooter ? excludeLastFooter : false,
+      autoScript: autoScript ? autoScript : undefined,
+    },
+  });
+
+  const onSubmit = async (data: AddProviderConfigFormData) => {
+    if (!isDirty) {
+      return closeOverlay();
+    }
+    const props = Object.entries(data.dynamicFields).map(([key, value]) => ({ name: key, value: value }));
+    const { providerName, dynamicFields, ...rest } = data;
+    const payload = {
+      ...rest,
+      props,
+      name: providerName,
+    };
+
+    await saveConfig([...providerConfigurations, payload as ProviderConfiguration])
+      .then(() => {
+        showSnackbar({
+          title: t('configSaved', 'Configuration saved'),
+          kind: 'success',
+        });
+        mutateConfigs();
+        closeOverlay();
+      })
+      .catch(() =>
+        showSnackbar({
+          title: t('configNotSaved', 'Configuration not saved'),
+          kind: 'error',
+        }),
+      );
+  };
+
+  const selectedTemplate = templates[watch('templateName')] ?? null;
```

Comment:
Won't this initially throw an exception?

```suggestion
  const selectedTemplate = templates?.[watch('templateName')] ?? null;
```